src/file.h

```diff
@@ -0,0 +1,30 @@
+#ifndef FILE_H
+#define FILE_H
+
+#include "kerneltypes.h"
+
+#define FILE_MODE_READ (1u << 0)
+#define FILE_MODE_WRITE (1u << 1)
+
+struct memory_inode_t
+{
+	const char *filename;
+	char *data_start;
+};
+
+struct file_t
+{
+	uint8_t mode;
+	uint32_t offset;
+	uint32_t inode;
+	bool in_memory;
+
+	int (*read)(struct file_t *, char *, uint32_t);
+	int (*write)(struct file_t *, char *, uint32_t);
+};
+
+int file_init();
+int register_memory_inode(const char *filename);
+int file_memory_inode_index(const char *filename);
+
```

Comment:
This is the right idea. 

I think it is a good practice to design the header file to reflect how you want it to work in the general case, even if it isn't yet implemented.  Can you think of the other methods that a file needs?  Perhaps we should have an enumeration for `type` instead of a boolean for `in_memory`.